### PR TITLE
Removing duplicate code

### PR DESCRIPTION
**Pull request by meiao**
### Overview
Removes duplicate code.
Adds some comments so we know that some things are used in the security agent.

### CODE REVIEW

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java [hunk 1]
```diff
@@ -249,24 +248,7 @@ private Collection<ClassMatchVisitorFactory> loadInternalSecurityWeavePackages()
...
-                                }
-                            } else {
-                                LOG.log(Level.FINER, "adding weave package: {0}", internalWeavePackage.getName());
-                                internalWeavePackages.add(internalWeavePackage.getName());
-                                weavePackageManager.register(internalWeavePackage);
-                            }
-                        } catch (Throwable t) {
-                            LOG.log(Level.FINER, t, "unable to load weave package jar {0}", instrumentationUrl);
-                        }
+                        registerInstrumentation(instrumentationUrl);
```
Comment (meiao) [inline newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java:251]: This region of code was identical to the original code that read the instrumentation modules. So extracted into a method and both use the same code.

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java [hunk 2]
```diff
@@ -218,17 +218,16 @@ private WeavePackageConfig createWeavePackageConfig(JarInputStream jarStream, St
     /**
      * Load all the security weave packages embedded in the agent jar.
      */
-    private Collection<ClassMatchVisitorFactory> loadInternalSecurityWeavePackages() {
+    private void loadInternalSecurityWeavePackages() {
```
Comment (meiao) [inline newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java:221]: This method returned a collection, which was instantiated and never populated.
Also the return value was always ignored. So removing the allocation and the return value.

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java [hunk 3]
```diff
@@ -300,9 +280,7 @@ public void registerInstrumentationCloseable(String instrumentationName, Closeab
     /**
      * Load all the weave packages embedded in the agent jar.
      */
-    private Collection<ClassMatchVisitorFactory> loadInternalWeavePackages() {
-        final Collection<ClassMatchVisitorFactory> matchers = Sets.newConcurrentHashSet();
-
+    private void loadInternalWeavePackages() {
```
Comment (meiao) [inline newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/weaver/ClassWeaverService.java:283]: This is the original method that got copied for the security agent. Same changes from there.

File: newrelic-agent/src/main/java/com/newrelic/bootstrap/BootstrapAgent.java [hunk 4]
```diff
@@ -113,21 +113,10 @@ public static void premain(String agentArgs, Instrumentation inst) {
         }
 
         checkAndApplyIBMLibertyProfileLogManagerWorkaround();
-        checkAndApplyJbossAdjustments(inst);
+        JbossUtils.checkAndApplyJbossAdjustments(inst);
```
Comment (meiao) [inline newrelic-agent/src/main/java/com/newrelic/bootstrap/BootstrapAgent.java:116]: See JBossUtil comment.

File: .github/actions/setup-environment/action.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 name: Setup Environment
 description:  Sets up the environment to build/test the agent.
-# This action expects the agent to be checked out at $GITHUB_WORKSPACE and Security agent at $GITHUB_WORKSPACE/k2-javaagent.
-# It will also set up the instrumentation jar zip if S3 credentials are set.
+# This action expects the agent to be checked out at $GITHUB_WORKSPACE.
```
Comment (meiao) [inline .github/actions/setup-environment/action.yml:3]: Since the security agent is in maven central, we no longer need to download it.

**Review by jtduffy — APPROVED** [review]

**Review by jasonjkeller — APPROVED** [review]

**Pull request merged**